apps/remote_control/lib/lexical/remote_control/build/error.ex

```diff
@@ -16,15 +16,74 @@ defmodule Lexical.RemoteControl.Build.Error do
   end
 
   def parse_error_to_diagnostics(context, message_info, token) do
-    diagnostic = %Diagnostic{
+    if missing_terminator?(message_info) do
+      build_missing_terminator_diagnostics(context, message_info)
+    else
+      diagnostic = %Diagnostic{
+        file: nil,
+        severity: :error,
+        position: context_to_position(context),
+        compiler_name: "Elixir",
+        message: message_info_to_binary(message_info, token)
+      }
+
+      [diagnostic]
+    end
+  end
+
+  defp build_missing_terminator_diagnostics(context, message_info) do
+    end_line_diagnositic = build_end_line_diagnostic(context, message_info)
+    start_line_diagnostic = build_start_line_diagnostic(message_info)
+    maybe_hint_diagnostic = build_hint_diagnostic(message_info)
+
+    Enum.reject([end_line_diagnositic, start_line_diagnostic, maybe_hint_diagnostic], &is_nil/1)
+  end
+
+  defp build_end_line_diagnostic(context, message_info) do
+    [end_line_message | _] = String.split(message_info, "\n")
+
+    %Diagnostic{
       file: nil,
       severity: :error,
       position: context_to_position(context),
       compiler_name: "Elixir",
-      message: message_info_to_binary(message_info, token)
+      message: end_line_message
+    }
+  end
+
+  @start_line_regex ~r/(\w+) \(for (.*) starting at line (\d+)\)/
+  defp build_start_line_diagnostic(message_info) do
+    start_line_result = Regex.run(@start_line_regex, message_info)
+    [_, missing, token, start_line] = start_line_result
+
+    %Diagnostic{
+      file: nil,
+      severity: :error,
+      position: String.to_integer(start_line),
+      compiler_name: "Elixir",
+      message: "The #{token} here is missing a terminator: #{inspect(missing)}"
     }
+  end
+
+  @hint_regex ~r/HINT: .*on line (\d+).*/m
+  defp build_hint_diagnostic(message_info) do
+    hint_result = Regex.run(@hint_regex, message_info)
```

Comment:
I tihnk this would be better represented via a case
```elixir 
case Regex.run(@hint_regex, message_info) do 
  [message, hint_line] -> 
      ...
      
  _ -> nil
end
```